src/ar_storage.erl

```diff
@@ -49,52 +50,94 @@ ensure_directories() ->
 	filelib:ensure_dir(filename:join(DataDir, ?WALLET_LIST_DIR) ++ "/"),
 	filelib:ensure_dir(filename:join(DataDir, ?HASH_LIST_DIR) ++ "/").
 
-%% @doc Clear the cache of saved blocks.
-clear() ->
-	lists:map(fun file:delete/1, filelib:wildcard(filename:join([ar_meta_db:get(data_dir), ?BLOCK_DIR, "*.json"]))),
-	ar_block_index:clear().
+count_blocks_on_disk() ->
+	spawn(
+		fun() ->
+			DataDir = ar_meta_db:get(data_dir),
+			case file:list_dir(filename:join(DataDir, ?BLOCK_DIR)) of
+				{ok, List} ->
+					ar_meta_db:increase(blocks_on_disk, length(List));
+				{error, Reason} ->
+					ar:warn([
+						{event, failed_to_count_blocks_on_disk},
+						{reason, Reason}
+					])
+			end
+		end
+	).
 
-%% @doc Removes a saved block.
-delete_block(Hash) ->
-	ar_block_index:remove(Hash),
-	file:delete(block_filepath(Hash)).
+rename_block_files(HL) ->
+	BlocksDir = filename:join(ar_meta_db:get(data_dir), ?BLOCK_DIR),
+	case filelib:is_file(filename:join(BlocksDir, "blocks_renamed")) of
```

Comment:
So do we track this migration with the existence of this `block_renamed` file? If so, we might wanna name the file something that indicates the new format, otherwise, we might have something like `blocks_renamed_again` in the future... Maybe something like `indep_hash_only_names`. I'm not sure if we'd wanna postfix some `_v1` afterthat, that might be overengineering it, but then again, [you can see how far I went with SQLite](https://github.com/ArweaveTeam/arweave-private/pull/148/files#diff-f1df8ff18cce849ef7902ab918e314a2R277) 🙃